Source/ASMultiplexImageNode.mm

```diff
@@ -271,49 +279,34 @@ - (BOOL)placeholderShouldPersist
 - (void)displayWillStartAsynchronously:(BOOL)asynchronously
 {
   [super displayWillStartAsynchronously:asynchronously];
-  
   [self didEnterPreloadState];
-  
-  if (_downloaderImplementsSetPriority) {
-    {
-      ASDN::MutexLocker l(_downloadIdentifierLock);
-      if (_downloadIdentifier != nil) {
-        [_downloader setPriority:ASImageDownloaderPriorityImminent withDownloadIdentifier:_downloadIdentifier];
-      }
-    }
-  }
+  [self _updatePriorityOnDownloaderIfNeededWithDefaultPriority:ASImageDownloaderPriorityImminent];
 }
 
 /* didEnterVisibleState / didExitVisibleState in ASNetworkImageNode has a very similar implementation. Changes here are likely necessary
  in ASNetworkImageNode as well. */
 - (void)didEnterVisibleState
 {
   [super didEnterVisibleState];
-  
-  if (_downloaderImplementsSetPriority) {
-    ASDN::MutexLocker l(_downloadIdentifierLock);
-    if (_downloadIdentifier != nil) {
-      [_downloader setPriority:ASImageDownloaderPriorityVisible withDownloadIdentifier:_downloadIdentifier];
-    }
-  }
-  
+  [self _updatePriorityOnDownloaderIfNeededWithDefaultPriority:ASImageDownloaderPriorityVisible];
   [self _updateProgressImageBlockOnDownloaderIfNeeded];
 }
 
 - (void)didExitVisibleState
 {
   [super didExitVisibleState];
-  
-  if (_downloaderImplementsSetPriority) {
-    ASDN::MutexLocker l(_downloadIdentifierLock);
-    if (_downloadIdentifier != nil) {
-      [_downloader setPriority:ASImageDownloaderPriorityPreload withDownloadIdentifier:_downloadIdentifier];
-    }
-  }
-  
+  [self _updatePriorityOnDownloaderIfNeededWithDefaultPriority:ASImageDownloaderPriorityPreload];
```

Comment:
Just reminder that in thrash case (without coalescing) a view may exit and enter immediately. It seems ok in this case though.

Reply:
Good call. I think the right fix is coalescing in such case to avoid jumping between interface states unnecessarily.